internal/olm/client/client.go

```diff
@@ -112,16 +112,52 @@ func NewClientForConfig(cfg *rest.Config) (*Client, error) {
 func (c Client) DoCreate(ctx context.Context, objs ...client.Object) error {
 	for _, obj := range objs {
 		kind := obj.GetObjectKind().GroupVersionKind().Kind
-		log.Infof("  Creating %s %q", kind, getName(obj.GetNamespace(), obj.GetName()))
-		err := c.KubeClient.Create(ctx, obj)
+		resourceName := getName(obj.GetNamespace(), obj.GetName())
+
+		log.Infof("  Creating %s %q", kind, resourceName)
+
+		if err := c.safeCreateOneResource(ctx, obj, kind, resourceName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// try to create 10 times before giving up
```

Comment:
Is there a reason for the 10?  I was going to ask about the 5, but since then it's become 10, and I wonder if it might indeed become something else in future, since we're dealing with unknown forces which cause an atomic step to fail.  
I like that you're solving the general case instead of the specific one, but would it be possible for you to define an environment variable which would be used for this value, and if not found would default to 10? 

Something like 
```golang
const resourceCreateRetryCountEnv = "SDK_RESOURCE_RETRY_COUNT"

func getResourceCreateRetryCount() int {
   if s := os.Getenv(resourceCreateRetryCountEnv); s != "" {
     if i, e := strconv.Atoi(s); e == nil {
       return i
     }
   }
   
   return 10
}
```

and then the for loop is 
```golang
max = getResourceCreateRetryCount()
for i:=0; i < max; i++ {
```